### ISSUE TITLE
Evaluating gradient field

### ISSUE THREAD

**Issue by kerrybardot** [Loop3D/LoopStructural#169]
Hello.

I am evaluating the scalar field gradient to calculate the direction of the 3D K tensor for a geological model. I'm a little confused about the output (probably because my 3D geometry isn't amazing). I'm assuming directional derivatives in the 3 directions of which I would only be interested in the first two. However the gradients are all negative which is curious as there are definitely some sections of the model where it slopes in the other direction... Do you have any more info for me on the scalar gradient output! 

Thank you.

**Comment by lachlangrose** [comment]
Hi @kerrybardot,

If you are calculating the gradient using `model.evaluate_feature_gradient(feature_name,locations)` this will return the derivative of the implicit function wrt x,y,z. The only things to be aware of is if you are using the model[feature_name].evaluate_value(locations), these locations need to be in the "model coordinates" which are reprojected so that the origin is 0. 

I would check the gradient is calculated correctly by visually assessing it.

view = LavaVuModelViewer(model)
view.add_isosurfaces(model[feature_name],nslices=10)
view.add_vector_field(model[feature_name],locations=model.regular_grid(shuffle=True)[::100]) # ::100 takes every 100th point and they are shuffled - this just helps with visualisation.
view.add_vector_field(model[feature_name],locations=model.regular_grid(shuffle=True)[::100],name='vector_disks',symbol_type='disk') #136 
view.interactive()

This should allow you to determine if the vector field matches your scalar field.

I can't think of any other things to consider, if this doesn't help can you share the model?

Good luck

**Comment by kerrybardot** [comment]
Thanks very much Lachlan for your help.

I am very excited about LOOP Structural and trying to link it with a groundwater model in Python. It is going well, for the most part!
The only thing I am struggling with is converting the gradient field. I need to find the gradient of the bedding to specify the conductivity vector. I have made a simple synthetic model with a fault:

***@***.***

The vector field that plot seems to be the inputted one (I just specified (0,0,1) as when you drill a hole you don't really know the dip until you join together the surfaces).
I thought that the vector field would match the implicit function.

My maths is not amazing, so maybe I have misunderstood something! I was expecting to see the 5 degree dip and some weird gradients around the fault...

It would be great to have a quick Teams or Zoom call if you had 10 min, or otherwise I can share model (I just have to get it looking OK first!)

Thanks
Kerry
p.s. in Perth, Australia, so going to bed shortly but can chat within next 30 minutes, or any day!


***@***.***

***@***.***
***@***.***
